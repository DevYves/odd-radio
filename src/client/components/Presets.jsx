import React, { Component } from 'react';

class Presets extends Component {
  constructor(props){
    super(props);
  }

  render() {
    return (
      <div className="container presets">
        <div className="center presets-title">
          { this.props.title }
        </div>
        <div className="row border">
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.presets.one)}>
            1: { this.props.presets.one.name ? this.props.presets.one.name : 'Empty' }
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.presets.two)}>
            2: { this.props.presets.two.name ? this.props.presets.two.name : 'Empty' }
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.presets.three)}>
            3: { this.props.presets.three.name ? this.props.presets.three.name : 'Empty'  }
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.presets.four)}>
            4: { this.props.presets.four.name ? this.props.presets.four.name : 'Empty' }
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.presets.five)}>
            5: { this.props.presets.five.name ? this.props.presets.five.name : 'Empty' }
          </div>
        </div>
      </div>
    );
  }
}

export default Presets;