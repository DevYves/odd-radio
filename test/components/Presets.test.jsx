import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import Presets from '../../src/client/components/Presets';

describe('Presets Component', () => {
    const props = {
      presets: {
        one: {name: 'CMYK'},
        two: {name: 'CMYK'},
        three: {name: 'CMYK'},
        four: {name: 'CMYK'},
        five: {name: 'CMYK'}
    },
      title:"Presets",
      clickFunction: jest.fn(),
      style: "audio-player-presets",
      stationFeed: {
        name: 'CMYK'
      }
    }

    const props2 = {
      presets: {
      one: {},
      two: {},
      three: {},
      four: {},
      five: {}
    },
    title:"Save Station in Presets",
    clickFunction: jest.fn(),
    details: {
       id: '1',
       name: 'QWER',
       audioFeed: 'www.google.ca',
       streamType: 'audio/mp3',
       frequency: '91.2 FM',
       city: 'Hamilton',
       province: 'ON'
     }

    }

  const subject = <Presets { ...props } />;
  const subject2 = <Presets { ...props2} />

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });

  describe('when preset one is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.five').at(0).simulate('click');

    expect(props.clickFunction).toHaveBeenCalled();
  })

  describe('when preset two is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.five').at(1).simulate('click');

    expect(props.clickFunction).toHaveBeenCalled();
  })

  describe('when preset three is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.five').at(2).simulate('click');

    expect(props.clickFunction).toHaveBeenCalled();
  })

  describe('when preset four is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.five').at(3).simulate('click');

    expect(props.clickFunction).toHaveBeenCalled();
  })

  describe('when preset five is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.five').at(4).simulate('click');

    expect(props.clickFunction).toHaveBeenCalled();
  })


});