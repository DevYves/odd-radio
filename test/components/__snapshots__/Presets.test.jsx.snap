// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presets Component matches snapshot 1`] = `
<div
  className="container presets"
>
  <div
    className="center presets-title"
  >
    Presets
  </div>
  <div
    className="row border"
  >
    <div
      className="five columns border-right undefined "
      onClick={[Function]}
    >
      <p>
        1: 
        CMYK
      </p>
    </div>
    <div
      className="five columns border-right undefined "
      onClick={[Function]}
    >
      <p>
        2: 
        CMYK
      </p>
    </div>
    <div
      className="five columns border-right undefined "
      onClick={[Function]}
    >
      <p>
        3: 
        CMYK
      </p>
    </div>
    <div
      className="five columns border-right undefined "
      onClick={[Function]}
    >
      <p>
        4: 
        CMYK
      </p>
    </div>
    <div
      className="five columns undefined "
      onClick={[Function]}
    >
      <p>
        5: 
        CMYK
      </p>
    </div>
  </div>
</div>
`;
